README.md

```diff
@@ -1154,6 +1157,127 @@ supported for various relational database flavors:
+    name: String
+}
+```
+is expected to correspond to a SQLAlchemy table object of the same name (case-insensitive) like
+
+```python
+from sqlalchemy import MetaData, Table, Column, String
+# table for GraphQL type Animal
+metadata = MetaData()
+animal_table = Table(
+    'animal', # name of table matches type name from schema
+    metadata,
+    Column('name', String(length=12)), # Animal.name field has corresponding 'name' column
+)
+```
+
+
+### End-To-End SQL Example
+An end-to-end example including relevant GraphQL schema and SQLAlchemy engine preparation follows.
+
+Note this is intended to show the setup steps for the SQL backend of the Graphql compiler, and does
+not represent best practices for configuring and running SQLAlchemy in a production system. Where
+possible links to relevant SQLAlchemy documentation are included.
+
+```python
+from graphql import parse
+from graphql.utils.build_ast_schema import build_ast_schema
+from sqlalchemy import MetaData, Table, Column, String, create_engine
+from graphql_compiler.compiler.ir_lowering_sql.metadata import SqlMetadata
+from graphql_compiler import compile_graphql_to_sql, insert_arguments_into_query
+
+# Step 1: Configure a GraphQL schema (note that this can also be done programmatically)
+schema_text = '''
+schema {
+    query: RootSchemaQuery
+}
+# IMPORTANT NOTE: all compiler directives are expected here, but not shown to keep the example brief
+
+directive @filter(op_name: String!, value: [String!]!) on FIELD | INLINE_FRAGMENT
+
+directive @output(out_name: String!) on FIELD
+
+type Animal {
+    name: String
+}
+'''
+schema = build_ast_schema(parse(schema_text))
+
+# Step 2: For all GraphQL types, bind the corresponding SQLAlchemy Table to a SQLAlchemy metadata
+# instance, using the expected naming detailed above.
+# See https://docs.sqlalchemy.org/en/latest/core/metadata.html for more details on this step.
+metadata = MetaData()
+animal_table = Table(
+    'animal', # name of table matches type name from schema
+    metadata,
+    # Animal.name schema field has corresponding 'name' column in animal table
+    Column('name', String(length=12)), 
+)
+
+# Step 3: Prepare a SQLAlchemy engine to query the target relational database.
+# See https://docs.sqlalchemy.org/en/latest/core/engines.html for more detail on this step.
+engine = create_engine('<connection string>')
+
+# Step 4: Wrap the SQLAlchemy metadata and dialect as a SqlMetadata GraphQL compiler object
+sql_metadata = SqlMetadata(engine.dialect, metadata)
+
+# Step 5: Prepare and compile a GraphQL query against the schema
+graphql_query = '''
+{
+    Animal {
+        name @output(out_name: "animal_name")
+             @filter(op_name: "in_collection", value:["$names"])
+    }
+}
+'''
+parameters = {
+    'names': ['animal name 1', 'animal name 2'],
+}
+
+compilation_result = compile_graphql_to_sql(schema, graphql_query, sql_metadata)
+compilation_result = insert_arguments_into_query(compilation_result, parameters)
+
+
```

Comment:
this is the only place where two consecutive new lines precede a new step, let's make it consistent one way or the other